Firefighting equipment

Flying-loulou opened the pull request:
:cl: Flying_loulou
tweak: Resprites the red O2 tank to make it look like an actual firefighter's tank.
tweak: The duty boots, jackboots and workboots are now fireproof.
tweak: The firesuit no longer covers the whole body, but only the chest, arms and groin.
rscadd: Adds a "fire overpant", to be worn over the uniform as an accessory to protect your legs against a fire.
rscadd: Adds the fire gloves, fully fire proof.
rscadd: Adds the EFT (Emergency Forcing Tool).
rscadd: Adds the "water grenade" to fight fires more efficiently.
rscadd: Creates the firebelt, which has 5 slots and can contain EFTs, water grenades, inflatable doors and mini-extinguishers. It comes filled with 3 water grenades, 1 EFT and 1 inflatable door.
rscadd: Adds the new equipment in the fire closets (places them into a dufflebag for convenience)
tweak: In order to survive a fire, you now have to wear the fire overpants, firesuit, fire gloves, duty boots (Jackboots and Work boots are fire resistant as well), as well as a fire resistant helmet/hardhat and proper internals.
/:cl:

A lil' PR to improve the firefighting equipment in the game. I'll probably keep going and try to work on fires themselves, to improve them as well, but that's for another day.

So now you have to equip a "fire overpant", as an accessory, as well as fire-proof gloves". The duty boots, jackboots and workboots are now fire-proof.
Creates an EFT (Emergency Forcing tool), which is basically a reworked and resprited crowbar (based on an existing tool), as well as water grenades, and creates a new "fire belt" to carry the equipment. It has 5 slots, and can hold the water grenades, EFTs, mini-extinguishers, or inflatable doors.
Places all the equipment described in a dufflebag, and places said dufflebag in the fire and hydrant closets.
I also resprited the red oxygen tank to make it look a bit better.

**Pictures**

The water grenade
![Fire_grenade](https://user-images.githubusercontent.com/49248697/66275537-ec0dc600-e889-11e9-80cb-34e24e2490ca.JPG)
![Fire_grenade_2](https://user-images.githubusercontent.com/49248697/66275538-eca65c80-e889-11e9-8cd8-9fb52aee5000.JPG)

The resprited tank
![fire_O2_tank_1](https://user-images.githubusercontent.com/49248697/66275539-eca65c80-e889-11e9-96fa-0285f882d372.JPG)
![fire_O2_tank_2](https://user-images.githubusercontent.com/49248697/66275540-eca65c80-e889-11e9-803e-d40ea0e5b024.JPG)
![fire_O2_tank_3](https://user-images.githubusercontent.com/49248697/66275541-eca65c80-e889-11e9-91a9-0983f5e7d0a2.JPG)
![fire_O2_tank_4](https://user-images.githubusercontent.com/49248697/66275542-ed3ef300-e889-11e9-9da0-44160800d046.JPG)

The overpants
![Fire_overpants1](https://user-images.githubusercontent.com/49248697/66275543-ed3ef300-e889-11e9-946b-b8d66cc44df0.JPG)
![Fire_overpants2](https://user-images.githubusercontent.com/49248697/66275544-ed3ef300-e889-11e9-9179-55ce004ea17f.JPG)
![Fire_overpants3](https://user-images.githubusercontent.com/49248697/66275545-edd78980-e889-11e9-8ba3-14e49179a9bd.JPG)
![Fire_overpants4](https://user-images.githubusercontent.com/49248697/66275546-edd78980-e889-11e9-9c40-d71f0685e8ad.JPG)

The jacket
![fire-jacket1](https://user-images.githubusercontent.com/49248697/66275547-edd78980-e889-11e9-8e20-e7a0d0b037a4.JPG)
![fire-jacket2](https://user-images.githubusercontent.com/49248697/66275548-edd78980-e889-11e9-8a99-5e263d3d94b3.JPG)
![fire-jacket3](https://user-images.githubusercontent.com/49248697/66275549-ee702000-e889-11e9-9109-4662c1345cc4.JPG)
![fire-jacket4](https://user-images.githubusercontent.com/49248697/66275550-ee702000-e889-11e9-9787-f5715def8c03.JPG)

The EFT (emergency forcing tool)
![Fire_EFT](https://user-images.githubusercontent.com/49248697/66275551-ee702000-e889-11e9-88b1-e9b9dd9e62fc.JPG)
![Fire_EFT-1](https://user-images.githubusercontent.com/49248697/66275552-ee702000-e889-11e9-8f64-0e74f9573462.JPG)
![Fire_EFT-2](https://user-images.githubusercontent.com/49248697/66275553-ef08b680-e889-11e9-8026-5c93b3901255.JPG)

The fire gloves
![Fire_gloves_1](https://user-images.githubusercontent.com/49248697/66310693-240a1d00-e90d-11e9-9ae7-c0ba76fc61fd.JPG)
![Fire_gloves_2](https://user-images.githubusercontent.com/49248697/66310690-240a1d00-e90d-11e9-9f75-99ca97891aca.JPG)
![Fire_gloves_3](https://user-images.githubusercontent.com/49248697/66310692-240a1d00-e90d-11e9-9e2a-c1883bc9d134.JPG)

PsiOmegaDelta commented on code/game/objects/items/weapons/grenades/chem_grenade.dm:
```diff
@@ -303,3 +303,25 @@
+	stage = 2
+	path = 1
+
+	New()
```
Relative pathing
Use `/obj/item/weapon/grenade/chem_grenade/water/Initialize()`

PsiOmegaDelta commented on code/game/objects/items/weapons/grenades/chem_grenade.dm:
```diff
@@ -303,3 +303,25 @@
+	path = 1
+
+	New()
+		..()
```
`. = ..()`

PsiOmegaDelta commented on code/game/objects/items/weapons/storage/belt.dm:
```diff
@@ -470,3 +470,26 @@
+		)
+
+/obj/item/weapon/storage/belt/fire_belt/full/New()
+	..()
```
```
/obj/item/weapon/storage/belt/fire_belt/full/Initialize()
	. = ..()
```

PsiOmegaDelta commented on code/game/objects/items/weapons/storage/belt.dm:
```diff
@@ -470,3 +470,26 @@
+	new /obj/item/weapon/crowbar/emergency_forcing_tool(src)
+	new /obj/item/weapon/extinguisher/mini(src)
+	new /obj/item/weapon/grenade/chem_grenade/water(src)
+	new /obj/item/weapon/grenade/chem_grenade/water(src)
```
```
/obj/item/weapon/storage/belt/fire_belt
    [...]
    starts_with = list(
        /obj/item/inflatable/door,
        /obj/item/weapon/crowbar/emergency_forcing_tool,
        /obj/item/weapon/extinguisher/mini,
        /obj/item/weapon/grenade/chem_grenade/water = 2
    )

Flying-loulou commented:
Changed a few things according to @PsiOmegaDelta's requests.
Also added a fire glove instead of making the duty glove fire-resistant, following some feedbacks on discord.

PsiOmegaDelta commented on code/game/objects/items/weapons/storage/backpack.dm:
```diff
@@ -236,6 +236,18 @@
+	new /obj/item/clothing/accessory/fire_overpants(src)
+	new /obj/item/weapon/tank/oxygen/red(src)
+	new /obj/item/clothing/head/hardhat/firefighter(src)
+	new /obj/item/weapon/extinguisher
```
In the type definition:
```
/obj/item/weapon/storage/backpack/dufflebag/firefighter
    [...]
    startswith = list(
    	/obj/item/weapon/storage/belt/fire_belt/full
	    /obj/item/clothing/suit/fire/firefighter
	    /obj/item/weapon/extinguisher
	    /obj/item/clothing/gloves/fire
	    /obj/item/clothing/accessory/fire_overpants
	    /obj/item/weapon/tank/oxygen/red
	    /obj/item/clothing/head/hardhat/firefighter
	    /obj/item/weapon/extinguisher
    )

PsiOmegaDelta commented on code/modules/clothing/under/accessories/clothing.dm:
```diff
@@ -318,3 +318,19 @@
+	max_pressure_protection = FIRESUIT_MAX_PRESSURE
+	armor = list(melee = 0, bullet = 0, laser = 10, energy = 10, bomb = 10, bio = 0, rad = 0)
+
+/obj/item/clothing/accessory/fire_overpants/New()
```
`/obj/item/clothing/accessory/fire_overpants/Initialize()`

PsiOmegaDelta commented on code/modules/clothing/under/accessories/clothing.dm:
```diff
@@ -318,3 +318,19 @@
+	armor = list(melee = 0, bullet = 0, laser = 10, energy = 10, bomb = 10, bio = 0, rad = 0)
+
+/obj/item/clothing/accessory/fire_overpants/New()
+	..()
```
`. = ..()`

Flying-loulou commented:
Hopefully that fixes all the remaining issues.

Flying-loulou commented:
For some reason it appears that now none of those clothing are fireproof, while they were before, during previous tests (all I just adjusted the flags).

Flying-loulou commented:
Sorted it all out, thanks to @ChaosAlpha's great help.
@PsiOmegaDelta this is ready for review.

PsiOmegaDelta commented:
Code seems fine, PR commit count is a bit overgrown

Flying-loulou commented:
> Code seems fine, PR commit count is a bit overgrown

Should be good now (unless you want it to be squashed a bit more).

Flying-loulou commented:
@PsiOmegaDelta Bump

Flying-loulou committed:
Adds the fire clothing, equipments, tools and accessory mechanics
